### ISSUE TITLE
Cant transfer Autocrypt key (Enigmail -> Deltachat)

### ISSUE THREAD

**Issue by g6094199** [deltachat/deltachat-android#279]
....sucessfully transfered the key via autocrypt from deltachat 0.10 to enigmail 1.9 weeks before. now i want to transfer the key back to deltachat....

deltachat 0.16.0
enigmail 2.0

**Steps to reproduce the problem**

1. enigmail 2.0 on Debian Thunderbird
2. send key via autocrypt (doesnt matter if you set a password to the key or not)
3. insert setup code
4. -> invalid setup code

**Debug logs**

If applicable, debug logs can be copied from within the Delta Chat app: 

> 03-27 12:03:41.852 I/DeltaChat( 4253): *************** ApplicationLoader.onCreate() ***************
03-27 12:03:41.856 I/DeltaChat( 4253): JNI_OnLoad() ...
03-27 12:03:41.856 I/DeltaChat( 4253): JNI_OnLoad() succeeded.
03-27 12:03:41.857 I/DeltaChat( 4253): JNI: s_init_globals()...
03-27 12:03:41.858 I/DeltaChat( 4253): T2: Attaching ok.
03-27 12:03:41.858 I/DeltaChat( 4253): T2: Job thread entered.
03-27 12:03:41.858 I/DeltaChat( 4253): T2: Job thread waiting for signal...
03-27 12:03:41.873 I/DeltaChat( 4253): T1: Opened "/data/user/0/com.b44t.messenger.beta/files/messenger.db" successfully.
03-27 12:03:41.890 I/DeltaChat( 4253): T1: Signal job thread to wake up...
03-27 12:03:41.890 I/DeltaChat( 4253): T2: Job thread checks for pending jobs...
03-27 12:03:41.891 I/DeltaChat( 4253): T2: Executing job #1, action 900...
03-27 12:03:41.894 I/DeltaChat( 4253): T2: Connecting to IMAP-server "imap993" via SSL...
03-27 12:03:41.926 I/DeltaChat( 4253): Display size: 1080 x 1920, DPI: 422.03 x 424.069
03-27 12:03:42.081 I/DeltaChat( 4253): T1: Chatlist created in 3.317 ms.
03-27 12:03:42.087 I/DeltaChat( 4253): *** LaunchActivity.onResume()
03-27 12:03:42.130 I/DeltaChat( 4253): T2: Connection to IMAP-server ok.
03-27 12:03:42.130 I/DeltaChat( 4253): T2: Login to IMAP-server as "odysseus_held_der_antike"...
03-27 12:03:42.134 I/DeltaChat( 4253): *** KeepAliveService.onCreate()
03-27 12:03:42.140 I/DeltaChat( 4253): *** KeepAliveService.onStartCommand()
03-27 12:03:42.199 I/DeltaChat( 4253): T2: IMAP-Login ok.
03-27 12:03:42.199 I/DeltaChat( 4253): T2: IMAP-Capabilities: IMAP4rev1 CHILDREN ENABLE ID IDLE LIST-EXTENDED LIST-STATUS LITERAL+ MOVE NAMESPACE QUOTA SASL-IR SORT SPECIAL-USE THREAD=ORDEREDSUBJECT UIDPLUS UNSELECT WITHIN
03-27 12:03:42.199 I/DeltaChat( 4253): T2: Starting IMAP-watch-thread...
03-27 12:03:42.203 I/DeltaChat( 4253): T3: Attaching ok.
03-27 12:03:42.203 I/DeltaChat( 4253): T3: IMAP-heartbeat-thread started.
03-27 12:03:42.204 I/DeltaChat( 4253): T4: Attaching ok.
03-27 12:03:42.204 I/DeltaChat( 4253): T4: IMAP-watch-thread started.
03-27 12:03:42.204 I/DeltaChat( 4253): T4: Fetching from all folders.
03-27 12:03:42.213 I/DeltaChat( 4253): T2: Job #1 done and deleted from database
03-27 12:03:42.213 I/DeltaChat( 4253): T2: Job thread waiting for signal...
03-27 12:03:42.319 I/DeltaChat( 4253): T4: 0 mails read from "Chats" with 0 errors.
03-27 12:03:42.393 I/DeltaChat( 4253): T4: 0 mails read from "Drafts" with 0 errors.
03-27 12:03:42.393 I/DeltaChat( 4253): T4: Folder "Entwürfe" ignored.
03-27 12:03:42.463 I/DeltaChat( 4253): T4: 0 mails read from "Gesendet" with 0 errors.
03-27 12:03:42.619 I/DeltaChat( 4253): T4: Receiving message INBOX/1161029470...
03-27 12:03:42.711 I/DeltaChat( 4253): T4: Message sent by another messenger (will be moved to Chats-folder).
03-27 12:03:42.712 I/DeltaChat( 4253): T4: Message has 1 parts and is moved to chat #12.
03-27 12:03:42.727 I/DeltaChat( 4253): T1: Chatlist created in 0.187 ms.
03-27 12:03:42.728 I/DeltaChat( 4253): T1: Chatlist created in 0.131 ms.
03-27 12:03:42.728 I/DeltaChat( 4253): T1: Message list for search "(null)" in chat #0 created in 0.021 ms.
03-27 12:03:42.775 I/DeltaChat( 4253): T4: Receiving message INBOX/1161029471...
03-27 12:03:42.829 I/DeltaChat( 4253): T4: Message sent by another messenger (will be moved to Chats-folder).
03-27 12:03:42.830 I/DeltaChat( 4253): T4: Message has 1 parts and is moved to chat #13.
03-27 12:03:42.943 I/DeltaChat( 4253): T1: Chatlist created in 1.394 ms.
03-27 12:03:42.944 I/DeltaChat( 4253): T1: Chatlist created in 0.087 ms.
03-27 12:03:42.944 I/DeltaChat( 4253): T1: Message list for search "(null)" in chat #0 created in 0.015 ms.
03-27 12:03:42.987 I/DeltaChat( 4253): T4: Receiving message INBOX/1161029472...
03-27 12:03:43.003 I/DeltaChat( 4253): T4: Message sent by another messenger (will be moved to Chats-folder).
03-27 12:03:43.004 I/DeltaChat( 4253): T4: Message has 1 parts and is moved to chat #13.
03-27 12:03:43.014 I/DeltaChat( 4253): T1: Chatlist created in 0.185 ms.
03-27 12:03:43.015 I/DeltaChat( 4253): T1: Chatlist created in 0.098 ms.
03-27 12:03:43.015 I/DeltaChat( 4253): T1: Message list for search "(null)" in chat #0 created in 0.015 ms.
03-27 12:03:43.054 I/DeltaChat( 4253): T4: Receiving message INBOX/1161029473...
03-27 12:03:43.071 I/DeltaChat( 4253): T4: Message has 1 parts and is moved to chat #10.
03-27 12:03:43.094 I/DeltaChat( 4253): T4: 4 mails read from "INBOX" with 0 errors.
03-27 12:03:43.094 I/DeltaChat( 4253): T1: Chatlist created in 2.094 ms.
03-27 12:03:43.096 I/DeltaChat( 4253): T1: Chatlist created in 0.257 ms.
03-27 12:03:43.096 I/DeltaChat( 4253): T1: Message list for search "(null)" in chat #0 created in 0.037 ms.
03-27 12:03:43.161 I/DeltaChat( 4253): T4: 0 mails read from "Papierkorb" with 0 errors.
03-27 12:03:43.200 I/DeltaChat( 4253): T4: Init lastseenuid and attach it to UIDVALIDITY for folder "Postausgang".
03-27 12:03:43.200 I/DeltaChat( 4253): T4: Folder "Postausgang" is empty.
03-27 12:03:43.200 I/DeltaChat( 4253): T4: 0 mails read from "Postausgang" with 0 errors.
03-27 12:03:43.339 I/DeltaChat( 4253): T4: Receiving message Sent/1161027749...
03-27 12:03:43.429 I/DeltaChat( 4253): T4: Message is a reply to a messenger message (will be moved to Chats-folder).
03-27 12:03:43.429 I/DeltaChat( 4253): T4: Message has 1 parts and is moved to chat #13.
03-27 12:03:43.445 I/DeltaChat( 4253): T1: Chatlist created in 0.204 ms.
03-27 12:03:43.446 I/DeltaChat( 4253): T1: Chatlist created in 0.174 ms.
03-27 12:03:43.447 I/DeltaChat( 4253): T1: Message list for search "(null)" in chat #0 created in 0.021 ms.
03-27 12:03:43.484 I/DeltaChat( 4253): T4: Receiving message Sent/1161027750...
03-27 12:03:43.502 I/DeltaChat( 4253): T4: Message has 1 parts and is moved to chat #10.
03-27 12:03:43.525 I/DeltaChat( 4253): T4: 2 mails read from "Sent" with 0 errors.
03-27 12:03:43.525 I/DeltaChat( 4253): T4: Folder "Trash" ignored.
03-27 12:03:43.526 I/DeltaChat( 4253): T1: Chatlist created in 1.078 ms.
03-27 12:03:43.527 I/DeltaChat( 4253): T1: Chatlist created in 0.151 ms.
03-27 12:03:43.528 I/DeltaChat( 4253): T1: Message list for search "(null)" in chat #0 created in 0.021 ms.
03-27 12:03:43.647 I/DeltaChat( 4253): T4: Receiving message Unbekannt/1161027864...
03-27 12:03:43.655 I/DeltaChat( 4253): T4: Message has 1 parts and is moved to chat #14.
03-27 12:03:43.685 I/DeltaChat( 4253): T4: 1 mails read from "Unbekannt" with 0 errors.
03-27 12:03:43.685 I/DeltaChat( 4253): T4: Folder "Unerwünscht" ignored.
03-27 12:03:43.685 I/DeltaChat( 4253): T1: Chatlist created in 1.231 ms.
03-27 12:03:43.686 I/DeltaChat( 4253): T1: Chatlist created in 0.183 ms.
03-27 12:03:43.687 I/DeltaChat( 4253): T1: Message list for search "(null)" in chat #0 created in 0.020 ms.
03-27 12:03:43.746 I/DeltaChat( 4253): T4: IDLE start...
03-27 12:03:47.248 I/DeltaChat( 4253): T1: Message list for chat #10 created in 0.503 ms.
03-27 12:04:41.876 I/DeltaChat( 4253): *** TimerReceiver.onReceive()
03-27 12:04:46.102 W/DeltaChat( 4253): T1: Cannot decrypt Autocrypt Setup Message.
03-27 12:05:53.108 I/DeltaChat( 4253): *** LaunchActivity.onPause()
03-27 12:05:53.588 I/DeltaChat( 4253): *** Screen on
03-27 12:05:53.633 I/DeltaChat( 4253): *** Screen off
03-27 12:06:20.042 I/DeltaChat( 4253): *** TimerReceiver.onReceive()
03-27 12:07:21.580 I/DeltaChat( 4253): *** TimerReceiver.onReceive()
03-27 12:08:21.597 I/DeltaChat( 4253): *** TimerReceiver.onReceive()
03-27 12:09:23.588 I/DeltaChat( 4253): *** TimerReceiver.onReceive()
03-27 12:10:25.587 I/DeltaChat( 4253): *** TimerReceiver.onReceive()
03-27 12:11:26.590 I/DeltaChat( 4253): *** TimerReceiver.onReceive()
03-27 12:12:27.586 I/DeltaChat( 4253): *** TimerReceiver.onReceive()
03-27 12:12:37.884 I/DeltaChat( 4253): *** Screen on
03-27 12:12:41.644 I/DeltaChat( 4253): *** LaunchActivity.onResume()
03-27 12:13:01.497 I/DeltaChat( 4253): T4: IDLE has data.
03-27 12:13:01.497 I/DeltaChat( 4253): T4: IMAP stream lost; we'll reconnect soon.
03-27 12:13:01.508 I/DeltaChat( 4253): T4: Disconnecting...
03-27 12:13:01.510 I/DeltaChat( 4253): T4: Disconnect done.
03-27 12:13:01.514 I/DeltaChat( 4253): T4: Connecting to IMAP-server "imap.:993" via SSL...
03-27 12:13:01.947 I/DeltaChat( 4253): T4: Connection to IMAP-server ok.
03-27 12:13:01.948 I/DeltaChat( 4253): T4: Login to IMAP-server as "o"...
03-27 12:13:02.041 I/DeltaChat( 4253): T4: IMAP-Login ok.
03-27 12:13:02.137 I/DeltaChat( 4253): T4: IDLE start...
03-27 12:13:45.801 I/DeltaChat( 4253): *** TimerReceiver.onReceive()
03-27 12:13:50.891 I/DeltaChat( 4253): *** LaunchActivity.onPause()
03-27 12:13:51.309 I/DeltaChat( 4253): *** Screen off
03-27 12:14:45.840 I/DeltaChat( 4253): *** TimerReceiver.onReceive()
03-27 12:15:11.033 I/DeltaChat( 4253): *** Screen on
03-27 12:15:14.406 I/DeltaChat( 4253): *** LaunchActivity.onResume()
03-27 12:15:21.125 I/DeltaChat( 4253): *** LaunchActivity.onPause()
03-27 12:16:10.595 I/DeltaChat( 4253): *** TimerReceiver.onReceive()
03-27 12:16:37.309 I/DeltaChat( 4253): *** Screen off
03-27 12:17:11.602 I/DeltaChat( 4253): *** TimerReceiver.onReceive()
03-27 12:18:09.678 I/DeltaChat( 4253): *** Screen on
03-27 12:18:23.483 I/DeltaChat( 4253): *** TimerReceiver.onReceive()
03-27 12:18:23.556 I/DeltaChat( 4253): *** Screen off
03-27 12:18:52.243 I/DeltaChat( 4253): *** Screen on
03-27 12:19:23.508 I/DeltaChat( 4253): *** TimerReceiver.onReceive()
03-27 12:20:23.516 I/DeltaChat( 4253): *** TimerReceiver.onReceive()
<

**Comment by r10s** [comment]
i can confirm this bug.

**Comment by r10s** [comment]
@g6094199 can you please have a look a the content of the message sent by Thunderbird in Thunderbird > View source - in my test, the mail was way too short, sth like:

```
Message-Id: UalrEpOjoB9Y4A4sX110OZwkZDZ-enigmail
Date: Tue, 27 Mar 2018 13:15:44 GMT
To: ...
From: ...
Autocrypt-Setup-Message: v1
Subject: Autocrypt-Einrichtungsnachricht
Content-type: multipart/mixed; boundary="oUCdP1yLBMfYlR3bpp0xmG3Eiknp6MD5n"

--oUCdP1yLBMfYlR3bpp0xmG3Eiknp6MD5n
Content-Type: text/plain

Diese Nachricht enthält alle Informationen, um Ihre Autocrypt-Einstellungen zusammen mit Ihrem geheimen Schlüssel sicher von Ihrem Originalgerät zu übertragen.

Um Ihr neues Gerät für Autocrypt einzurichten, folgen Sie bitte den Anweisungen, die von Ihrem neuen Gerät angezeigt werden sollten.

Sie können diese Nachricht behalten und als Sicherungskopie für Ihren geheimen Schlüssel verwenden. Wenn Sie das tun möchten, sollten Sie das Passwort notieren und es sicher verwahren.
--oUCdP1yLBMfYlR3bpp0xmG3Eiknp6MD5n
Content-Type: application/autocrypt-setup
Content-Disposition: attachment; filename="autocrypt-setup-message.html"

<html><body>
<p>Dies ist die Autocrypt-Einrichtungsdatei, die für die Übertragung von Einstellungen und Schlüsseln zwischen Clients verwendet wird. Sie können sie mit dem Einrichtungscode entschlüsseln, den Sie auf Ihrem alten Gerät finden, und dann den enthaltenen Schlüssel in Ihren Schlüsselbund importieren.</p>
<pre>
-----BEGIN PGP MESSAGE-----
Passphrase-Format: numeric9x4
Passphrase-Begin: 19

wy4ECQMI/0pC+95jpBFg6koZcZISOKVI61iy3RytXcLGvF8uiMaBEJNp0FNm
aX760joBEQw6/c5P2FU5EVn+3LIIHr0nEIhjjZKBruDI/7mGFhaDlpcqmjY3
7S0GatJUaC1aOll32kxcU3jJ
=FhgV
-----END PGP MESSAGE-----
</pre></body></html>
--oUCdP1yLBMfYlR3bpp0xmG3Eiknp6MD5n--
```

... the part between `-----BEGIN PGP MESSAGE-----` and `-----END PGP MESSAGE-----` should normally read about 100 lines ... maybe more and enigmail issue.

**Comment by r10s** [comment]
just trying to decrypt the message using `gpg` - as expected, decryption works but enigmails payload is no private key but just nothing (zero bytes)

**Comment by g6094199** [comment]
so this is the source of the send mail

> Message-Id: IAG8a5XCF3Ev1Ac0T4FFIf95B5C-enigmail
Date: Tue, 27 Mar 2018 10:24:56 GMT
To: xxx@xxxde
From: xxx@xx.de
Autocrypt-Setup-Message: v1
Subject: Autocrypt-Einrichtungsnachricht
Content-type: multipart/mixed; boundary="qtMGLDQasdadsasdasdwoT77FcQPkxa"

--qtMGLDQasdasdasdasdasdQPkxa
Content-Type: text/plain

Diese Nachricht enthält alle Informationen, um Ihre Autocrypt-Einstellungen zusammen mit Ihrem geheimen Schlüssel sicher von Ihrem Originalgerät zu übertragen.

Um Ihr neues Gerät für Autocrypt einzurichten, folgen Sie bitte den Anweisungen, die von Ihrem neuen Gerät angezeigt werden sollten.

Sie können diese Nachricht behalten und als Sicherungskopie für Ihren geheimen Schlüssel verwenden. Wenn Sie das tun möchten, sollten Sie das Passwort notieren und es sicher verwahren.
--qtMGLDQlfvoT77FcQPkxa
Content-Type: application/autocrypt-setup
Content-Disposition: attachment; filename="autocrypt-setup-message.html"

<html><body>
<p>Dies ist die Autocrypt-Einrichtungsdatei, die für die Übertragung von Einstellungen und Schlüsseln zwischen Clients verwendet wird. Sie können sie mit dem Einrichtungscode entschlüsseln, den Sie auf Ihrem alten Gerät finden, und dann den enthaltenen Schlüssel in Ihren Schlüsselbund importieren.</p>
<pre>
-----BEGIN PGP MESSAGE-----
Passphrase-Format: numeric9x4
Passphrase-Begin: 01

wy4ECQMIrm1fasasdasdadsasd98yQfDaBlZIImW80y6jg4yj
1zcq0joBpkuCxwSadsadssadasdadsYT8fcIMWAtBK6+JsweciNvg2aG61UR
rExL8ObkQHvWvduo9ZVcIM37
=cyK8
-----END PGP MESSAGE-----
</pre></body></html>
--qtMGLDasdasdasdasdasdasdcQPkxa--
>

**Comment by r10s** [comment]
it may have something to do with an unset password in enigmail.

i tried again and were prompted four (!) times to type a password, after that the payload was larger, but still unencryptable. _might_ be that this key itself was protected with an additional password then ...

**Comment by g6094199** [comment]
i've always been asked 2 times for setting a password. i'm i dialogue with patrick brunswick on the sf bugtracker now.

**Comment by kevincox** [comment]
I had the same problem. The source of the message was much larger as you indicated it should be however it still failed to import with basically no info provided. I retyped the code many times and even tried generating the message twice.

```
22:01:50 T1: Cannot decrypt Autocrypt Setup Message.
```

**Comment by r10s** [comment]
@kevincox does your key in the setup message contain is protected by an additional key? Delta Chat does not support these keys, however, maybe we should improve the error message in this case.

you can check the exported private key manually by `gpg --decrypt autocrypt-setup-file.html` where the file is the file attached to the setup message generated by enigmail

**Comment by kevincox** [comment]
Hmm, it appears that the message is not valid:

```
% gpg --decrypt Downloads/autocrypt-setup-message.html
gpg: unknown armor header: Passphrase-Format: numeric9x4
gpg: unknown armor header: Passphrase-Begin: 84
gpg: AES256 encrypted session key
gpg: encrypted with 1 passphrase
gpg: decryption failed: Invalid cipher algorithm
```

I guess I should forward to enigmail.

**Comment by r10s** [comment]
> I guess I should forward to enigmail.

please do.

might also be that the import/export works when hitting OK instead of CANCEL while not entering a passphrase, see https://sourceforge.net/p/enigmail/bugs/772/#1814
i have not tested this yet, if anyone knows, please let us know :)

**Comment by JonasDralle** [comment]
me too! in fact, every setup mail is displayed twice
maybe that contributes to the non-working of the transfer?

**Comment by r10s** [comment]
is the setup messages only displayed twice or also sent twice?

**Comment by JonasDralle** [comment]
Enigmail sends it only once. Yet it appears twice in the chat with 'Me'. I assume because it's both in my 'Sent' and my 'Inbox'?
Using Thunderbird 52.7.0 (32-Bit) with Enigmail 2.0.3

I tried with both messages and tripple-checked the security codes. Always failed. I then re-send the setup message and did that again so basically I entered some obscure non-qr-based code 2*2*3=12 times!
Thought I was going insane when a friend of mine told me he had the same problem D.C.

**Comment by g6094199** [comment]
had exaclty the same errors as kollokollo with triing others way to import the key.

**Comment by Ampli-fier** [comment]
It could be that Android does not provide the information about the contents of the Download folder correctly to Delta Chat (actually to all apps). Have you tried to restart your device? This helped for [this issue](https://github.com/deltachat/deltachat-core/issues/50), reported on Delta Chat Core.

**Comment by d24phant** [comment]
Still have the same problem with deltachat, but it works with k9. key has a pw

**Comment by kollokollo** [comment]
Hm, i cannot See how IT World Workshop with k9 .

**Comment by kollokollo** [comment]
I cannot see how it would work with k9. How would you create an autocrypt setup message?

**Comment by kollokollo** [comment]
I think the problem is as follows:
* Enigmail cannot export the private key into a file when the password is set to empty. In this case the file contains only the public key and if you try to generate the autocrypt message, its the same, no privkey in it and the shown passphrase is 1234-1234-1234-1234-...
* If you use a pasword for exporting (or generating the autocrypt message), then a private key is exported, but delta-chat cannot read it, because it assume an empty password.

**Comment by webratte** [comment]
Hi,
I have a (stupid?) question to the devs.

Why is it not possible to allow a password?
Is there no way to store the password (maybe as option)?

So it would be much easier to import already existing keys.

**Comment by kollokollo** [comment]
My suggestion would be to use the only password delta chat already knows: The email-account password. This could be used by default, so delta chat would not need to ask for any or bother the user in any other way. And with enigmail one can more safe export the private key to a file. It would just be a matter of instructions: "Always use the email account password when asked to use one".

**Comment by testbird** [comment]
>  instructions: "Always use the email account password when asked to use one"

Isn't that rather a bad practice, to better instruct against.

The advice could be to use a temporary password for transfer-only usages, to protect the file until it really gets deleted after import.

**Comment by happyreacer** [comment]
By the way, In my solution it works in k-9 mail with import passphrase in the openkeychain app

**Comment by kollokollo** [comment]
Any progress here? The new version still does not find the key in an .asc file in /mnt/sdcard/Download.

**Comment by kollokollo** [comment]
I tried already entering an empty password. It did not help. Could you explain how you did it in detail?

**Comment by mazilaaa** [comment]
If you do not succeed I can make a detailed instruction. Then I will need to install the English version.

**Comment by mazilaaa** [comment]
I used Autocrypt and sent a message from my mail to my mail.
![default](https://user-images.githubusercontent.com/42761952/44707459-c59a2a80-aace-11e8-92df-ebcc4b703821.png)

**Comment by kollokollo** [comment]
![a](https://user-images.githubusercontent.com/15250834/44707716-d168fb00-aa94-11e8-9f91-d9c0e15eb931.png)
Thats what I get :-(

**Comment by mazilaaa** [comment]
I think I should make a detailed instruction. I also suffered for a long time. )))

**Comment by r10s** [comment]
@mazilaaa that would be great :)

**Comment by gorgabal** [comment]
Please be aware that using GPG without a passphrase is risky. Removing the passphrase is your own choice.

Is anybody currently working on a patch for this? I am willing and able to help out.

**Comment by mazilaaa** [comment]
I apologize, I recorded the video in Russian. But I think as an example it will be clear. https://youtu.be/R1uAKNfINww

**Comment by kollokollo** [comment]
@mazilaaa This is a nice tutorial. I think the general procedure is clear now. And obviously it work for you. However, enigmail doesnt on my computer (Ubuntu 16). Maybe I need to wait for an update. It looks like it is a bug in enigmail. I had hoped, that someone found a workaround.

**Comment by CoWinkKeyDinkInc** [comment]
I've had this problem for a while now, and I don't think it's anything that Delta Chat can fix, as it's up to Enigmail to resolve. I've managed to work around it by locally importing the keys to my phone and then getting it setup that way. My key pair that I generated with Enigmail on Thunderbird still works after being imported to Delta Chat and K9 Mail. Enigmail is doing something wrong when it comes to importing private keys and using autocrypt because so far its the only one that has problems importing keys. It also doesn't recognize mail that it should be able to decrypt after importing the private key. It's a problem on Enigmail but I have no idea where to go to tell them that (they still use Sourceforge, eww) I asked for help on their mailing list, but nobody has responded (I don't know if I was filtered out for my somewhat offensive email domain-name)

Since it seems the problem has been going on for so long, I think the only solution for me is to downgrade Enigmail to transfer the keys right and decrypt emails that I already have the key for. I wish there was another PGP add-on for Thunderbird so this wouldn't be so big of a problem. I got a new desktop and it's completely frustrating not being able to do what should be a simple transfer when I really want to move on from my really old laptop.

**Comment by r10s** [comment]
closing this for now, we can re-open as needed

**Comment by g6094199** [comment]
Tested:
Transfer Autocrypt Key from Enigmail 2.09 -> Deltachat 0.100 on Android AND Linux. Both worked successfully!

**Comment by r10s** [comment]
@g6094199 great - thanks for testing :+1:

**Comment by gessel** [comment]
I'm having this problem with Enigmail 2.1.9 (20201220-1657) --> Delta Chat 1.20.2.  I successfully imported the Enigmail generated Autocrypt Setup Message into K9 and FairEmail (via OpenKeyChain 5.6 and 5.7 respectively... gotta update one phone).  

The Delta Chat interface correctly shows the two "prompt" pin numbers (verified by repeating the Autocrypt Setup Message - Pin prompt properly updated). 

I've tried both the Delta Chat desktop client (Ubuntu 64 bit Version 1.15.5 (git: v1.15.4-1-g77948e7b)) and the f-Droid supplied client (1.20.2) and both give identical results: 

Autocrypt Setup Message
Bad setup code. Please Try again.

**Comment by Hocuri** [comment]
@gessel Thanks for your feedback!

However, this is a closed issue from 2 years ago. Could you please open a new issue, so that it does get attention? You can post a link to this issue here then ("There was a similar issue 2 years ago")

**Comment by gessel** [comment]
@Hocuri, thanks for the advice and done.
